constraints/src/main/java/gov/nasa/jpl/aerie/constraints/model/LinearEquation.java

```diff
@@ -0,0 +1,180 @@
+package gov.nasa.jpl.aerie.constraints.model;
+
+import gov.nasa.jpl.aerie.constraints.time.Interval;
+import gov.nasa.jpl.aerie.constraints.time.IntervalMap;
+import gov.nasa.jpl.aerie.constraints.time.Windows;
+import gov.nasa.jpl.aerie.merlin.protocol.types.Duration;
+import org.apache.commons.lang3.tuple.Pair;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.function.BiFunction;
+
+import static gov.nasa.jpl.aerie.constraints.time.Interval.Inclusivity.Exclusive;
+import static gov.nasa.jpl.aerie.constraints.time.Interval.Inclusivity.Inclusive;
+import static gov.nasa.jpl.aerie.constraints.time.Interval.at;
+import static gov.nasa.jpl.aerie.constraints.time.Interval.interval;
+
+/**
+ * A linear equation in point-slope form.
+ */
+public final class LinearEquation {
+  public final Duration initialTime;
+  public final double initialValue;
+  public final double rate;
+
+  public LinearEquation(final Duration initialTime, final double initialValue, final double rate) {
+    this.initialTime = initialTime;
+    this.initialValue = initialValue;
+    this.rate = rate;
+  }
+
+  public double valueAt(final Duration time) {
+    return this.initialValue + this.rate*(time.minus(this.initialTime)).ratioOver(Duration.SECOND);
+  }
+
+  public LinearEquation shiftInitialTime(final Duration newInitialTime) {
+    return new LinearEquation(
+        newInitialTime,
+        this.initialValue + newInitialTime.minus(this.initialTime).ratioOver(Duration.SECOND)*this.rate,
+        this.rate
+    );
+  }
+
+  public boolean changing() {
+    return this.rate != 0;
+  }
+
+  private Optional<Duration> intersectionPointWith(final LinearEquation other) {
+    if (this.rate == other.rate) return Optional.empty();
+    return Optional.of(
+        this.initialTime.plus(
+            Duration.roundNearest(
+                (other.valueAt(this.initialTime) - this.initialValue) / (this.rate - other.rate)*1000000,
```

Comment:
Why do we need a `1000000` scaling factor here? Could we instead use a different unit on the following line?

If we do keep the scaling factor, it would help my understanding if we could add parentheses to this expression - I'm not sure off the top of my head whether this is `a / (b * c)` or `(a / b) * c`.

Reply:
The rate parameter has the units `1/second`. I'll try to make it a little more readable, and use `Duration.SECOND` instead